File: client/src/app/_dtos/localUserDtos/userParams.ts
import { LocalUserDto } from "./localUserDto";

export class UserParams {
    pageNumber: number = 1;
    gender: string = "";
    minAge: number = 16;
    maxAge: number = 99;

    constructor(user: LocalUserDto) {
        switch (user.gender) {
            case 'male':
                this.gender = 'female';
                break;
            case 'female':
                this.gender = 'male';
                break;
            default:
                this.gender = user.gender;
        }
    }
}

File: src/DatingLoveApp.Business/Dtos/AppUsers/UserParams.cs
using DatingLoveApp.Business.Common.Constants;

namespace DatingLoveApp.Business.Dtos.AppUsers;

public class UserParams
{
    public string Gender { get; set; } = GenderConstants.Unknown;

    public int MinAge { get; set; } = 16;

    public int MaxAge { get; set; } = 99;

    public string SortBy { get; set; } = UserSortConstants.LastActive;

    public int PageNumber { get; set; } = 1;

    public int PageSize { get; set; } = 6;

    public override string ToString()
    {
        return $"{Gender}-{MinAge}-{MaxAge}-{SortBy}-{PageNumber}-{PageSize}";
    }
}
